File: python/PRIME1.py
# @Failed
import math
MAX=100005

def sieveOfEratosthenes():
    arr=[True for i in range(MAX)]
    arr[0]=False
    arr[1]=False
    i=2
    
    for i in range(2,int(math.sqrt(MAX))):
        if arr[i]:
            for j in range(2*i,MAX,i):
                arr[j]=False
    return arr

def operate(a,b,arr):
    temp=''
    for i in range(a,b+1):
        if arr[i]:
            temp+=str(i)
            temp+='\n'
    print(temp)
    print()


if __name__=='__main__':
    t=int(input())
    arr=sieveOfEratosthenes()
    while t:
        a=int(input())
        b=int(input())
        operate(a,b,arr)
        t=t-1

File: cpp/PRIME1.cpp
// Most efficient way to find prime numbers upto a ultra large numbers
// AKA Sieve of Eratosthenes
#include<bits/stdc++.h>

using namespace std;

#define MAX 1000000009

vector<bool> sieveOfEratosthenes()
{
    vector<bool> arr(MAX+1,true); // 1 signifies all prime
    arr[0]=false;
    arr[1]=false;
    //Actual Operation
    for(int i=2;i<=MAX;i++)
    {
        if(arr[i])
        {
            for(int j=2*i;j<=MAX;j+=i)
            {
                arr[j]=false;
            }
        }
    }
    return arr;
}
void operate(const int a,const int b,const vector<bool>& arr)
{
    
    for(int i=a;i<=b;i++)
    {
        if(arr[i])
        cout<<i<<endl;
    }
    cout<<endl;
}
int main()
{

    int t,a,b;
    cin>>t;
    vector<bool> arr=sieveOfEratosthenes();
    while(t--)
    {
        cin>>a>>b;
        operate(a,b,arr);
    }
    return 0;
}